### ISSUE TITLE
sstable mx reader: implement position fast forwarding in reverse reads

### ISSUE THREAD

**Issue by kbr-** [scylladb/scylla#9427]
This issue assumes that #9281 is merged into `master` (at the moment of writing the issue it is in `next`).

After #9281, when the sstable reader is created with a reversed slice and `streamed_mutation::forwarding::yes`, it will use the `make_forwardable` wrapper which is inefficient (`fast_forward_to(position_range)` simply consumes and drops fragments from the underlying reader until the desired range is reached).

As a follow-up, efficient forwarding should be implemented directly in the mx reader.

Refs: #9134

**Comment by avikivity** [comment]
Not a bug, not backporting.
